Backports v0.22.2

haampie opened the pull request:
- [x] Set version to v0.22.2.dev0
- [x] #45090
- [x] #45205
- [x] #45191
- [x] #46206
- [x] #45018
- [x] #45127
- [x] #45268
- [x] #45269
- [x] #45326
- [x] #45766
- [x] #46453
- [x] #46475
- [x] #45275
- [x] #45160
- [x] #45235
- [x] #45641
- [x] #45645
- [x] #45683
- [x] #42780
- [x] #45793
- [x] #46480
- [x] Set version to v0.22.2 and update CHANGELOG.md

**Packages**
- [x] #42197
- [x] #44931
- [x] #45304

arezaii committed:
Chapel package: major update (#42197)

* add cray detection taken from upcxx
* add CUDA/ROCm support
* add numerous pass-through options to Chapel build,
  like gpu_mem_strategy, comm_substrate, etc.; all variants are
  translated to analogous CHPL_* environment variables. As a side
  effect, this defines a number of environment variables that are
  not actually used by Chapel.
* Define LD_LIBRARY_PATH, LIBRARY_PATH, and PKG_CONFIG_PATH to
  help programs built with Chapel properly locate needed runtime
  dependencies

---------

Co-authored-by: bonachea <dobonachea@lbl.gov>

arezaii committed:
update chapel package for v2.1 (#44931)

adamjstewart committed:
Add support for macOS Sequoia (#45018)

haampie committed:
spec.py: fix __getitem__ looking outside of dag (#45090)

`Spec.__getitem__` queries dependent edges, which almost always point to
nodes outside the sub-dag considered. It should only ever look at edges
being traversed.

haampie committed:
Set version to v0.22.2.dev0

alecbcs commented:
Adding https://github.com/spack/spack/pull/45018 which is in the milestone, but want to make sure it doesn't get lost.

alalazo committed:
Add compatibility of sequoia with previous macOS versions (#45127)

* Add compatibility of sequoia with previous macOS versions

* Add compatibility of sequoia with previous macOS versions

haampie committed:
Avoid duplicate detectable tag (#45160)

in case of inheritance the static tags prop may be updated multiple
times, and it turns out builder classes magically inherit from
traditional package classes

haampie committed:
Add pkg- prefix to builtin.mock a b c d ... (#45205)

haampie committed:
Add c to the list of languages (#45191)

haampie committed:
unparser.py: remove print statements (#45235)

alalazo committed:
Run minimization of weights only on known targets (#45269)

This prevents excessive output from clingo of the kind:

.../spack/lib/spack/spack/solver/concretize.lp:1640:5-11: info: tuple ignored:
  #sup@2

alalazo committed:
Do not initialize previous store state in "use_store" (#45268)

The "use_store" context manager is used to swap the value
of a global variable (spack.store.STORE), while keeping
another global variable consistent (spack.config.CONFIG).

When doing that it tries to evaluate the previous value
of the store, if that was not done already. This is wrong,
since the configuration might be in an "intermediate" state
that was never meant to trigger side effects.

Remove that operation, and add a unit test to
prevent regressions.

arezaii committed:
Chapel package: updates post release (#45304)

* Fix +rocm variant, to ensure correct dependencies on ROCm packages
  and use of AMD LLVM
* Add a +pshm variant for comm=gasnet to enable fast shared-memory
  comms between co-locales
* Add logic to ensure we get the native CXI libfabric network provider
  on Cray EX
* Expand dependency type for package modules to encompass runtime
  dependencies
* Factor logic for setting (LD_)LIBRARY_PATH and PKG_CONFIG_PATH of
  runtime dependencies
* Workaround issue #44746 that causes a transitive dependency on lua
  to break SLURM
* Disable nonfunctional checkChplDoc test
* Annotate some variants as conditional, to improve spack info output
  and reduce confusion

---------

Co-authored-by: Dan Bonachea <dobonachea@lbl.gov>

bonachea commented:
Please merge backport PR #45514 to this branch for v0.22.2

alalazo committed:
Do not halt concretization on unknown variants in externals (#45326)

* Do not halt concretization on unkwnown variants in externals

dominichofer committed:
Remove execution permission from setup-env.sh (#45641)

`setup-env.sh` is meant to be sourced, not executed directly.
By revoking execution permissions, users who accidentally execute
the script will receive an error instead of seeing no effect.

* Remove execution permission from `setup-env.sh` and friends
* Don't make output file executable in `spack commands --update-completion`

---------

Co-authored-by: Todd Gamblin <tgamblin@llnl.gov>
Signed-off-by: Todd Gamblin <tgamblin@llnl.gov>

haampie committed:
buildcache: fix hard-coded, outdated layout version (#45645)

haampie committed:
deal with TimeoutError from ssl.py (#45683)

alalazo committed:
Update archspec to v0.2.5-dev (7e6740012b897ae4a950f0bba7e9726b767e921f) (#45721)

haampie committed:
Revert "Change environment modifications to escape with double quotes (#36789)" (#42780)

This reverts commit 690394fabc29908bbc4188ec29fd187fbb3ba13b, as it causes arbitrary code execution.

alalazo committed:
Run unit test in parallel again in CI (#45793)

The --trace-config option was failing for linux unit-tests,
so we were running serial.

becker33 committed:
include_concrete: read from older env formats properly (#45766)

* include_concrete: read from older env formats properly
* spack env rm: fix logic for checking env includes
* regression test

Signed-off-by: Todd Gamblin <tgamblin@llnl.gov>

haampie committed:
build_environment: explicitly disable ccache if disabled (#45275)

Jordan474 committed:
Fix regression in spec format string for indiviual variants (#46206)

Fix a regression in {variants.X} and {variants.X.value} spec format strings.

alalazo committed:
Bump archspec to latest commit (#46445)

This should fix an issue with Neoverse XX detection

Signed-off-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

haampie committed:
spack.util.url: fix join breakage in python 3.12.6 (#46453)

haampie committed:
docs: refer to upstreams.yaml in chain.rst title (#46475)

haampie committed:
run-unit-tests: no xdist if coverage (#46480)

xdist only slows down unit tests under coverage

haampie committed:
url join: fix oci scheme (#46483)

* url.py: also special case oci scheme in join

* avoid fetching keys from oci mirror

haampie committed:
require spec in develop entry (#46485)

haampie committed:
Set version to v0.22.2
